File: bioprocs/scripts/utils/runcmd.r
if (!exists('runcmd')) {
	runcmd = function (cmd, quit = TRUE, intern=FALSE) {
		write (paste(rep('-', 80), collapse="") , stderr())
		write ("Running command: ", stderr())
		write (cmd, stderr())
		
		if (!intern) {
			rc = try(system (cmd))
			if (rc == 0) {
				write ("Return code: 0", stderr())
				write (paste(rep('-', 80), collapse="") , stderr())
			} else {
				write (paste0("Return code: ", rc), stderr())
				write (paste(rep('-', 80), collapse="") , stderr())
				if (quit) {
					stop ('Quitted.')
				}
			}
			return (rc)
		} else {
			return (system(cmd, intern=T))
		}
	}
}

File: bioprocs/scripts/utils/runcmd.py
import sys
# https://github.com/kennethreitz/delegator.py
import delegator

class RuncmdException(Exception):
	pass

def runcmd(cmd, quit = True):
	c = delegator.run(cmd)
	sys.stderr.write('Running command at PID: %s\n' % c.pid)
	sys.stderr.write(cmd + '\n')
	c.block()
	sys.stderr.write('Return code: %s\n' % c.return_code)
	sys.stderr.write ('-'*80 + '\n')
	if quit and c.return_code != 0:
		raise RuncmdException('Command failed to run.')
	return c.return_code == 0
